### PR TITLE
Feature/repository links for webpages

### LINKED ISSUE #855: Add a repository link at the bottom of each webpage available on the dashboard site
### Issue Description
On the [Dashboard](https://dashboard.realdevsquad.com/) site of RDS, where links to various web pages are present at the bottom of each web page add the repository link of respective web page. 

### Expected Behavior
Every web page should have the repository link of its source code at the bottom.

### Current Behavior
All the web pages do not have its respective repository link at the bottom of page.

### Screenshots
Groups web page

### Reproducibility

 - [x] This issue is reproducible
 - [ ] This issue is not reproducible

### Steps to Reproduce
1. Visit the dashboard site.
2. Open any of the following web page: [`Discord Users`](https://dashboard.realdevsquad.com/users/discord/)/[`Discord Groups`](https://dashboard.realdevsquad.com/groups/)/[`Applications`](https://dashboard.realdevsquad.com/applications/)
3. Scroll to the bottom of the web page you will not find any repository link.

### Severity/Priority
 - [ ] Critical
 - [ ] High
 - [ ] Medium
 - [x] Low

### Checklist

- [x] I have read and followed the project's code of conduct.
- [x] I have provided all the necessary information to understand and reproduce the issue.

### PR DESCRIPTION
**Pull request by shantanu-02**
Date: 29/09/2024
Developer Name: shantanu-pawar-1

---

## Issue Ticket Number
- Closes #855

## Description
On the [Dashboard](https://dashboard.realdevsquad.com/) site of RDS, where links to various web pages are present at the bottom of each web page add the repository link of respective web page.

- Change 1: Added footer containing repository link for [Discord Users](https://dashboard.realdevsquad.com/users/discord/) web page.
- Change 2: Added footer containing repository link  for [Discord Groups](https://dashboard.realdevsquad.com/groups/) web page.
- Change 3: Added footer containing repository link  for [Applications](https://dashboard.realdevsquad.com/applications/) web page.
- 
### Under Feature Flag

- [ ] Yes
- [x] No

### Database Changes

- [ ] Yes
- [x] No

### Development Tested?

- [x] Yes
- [ ] No

## Screenshots
Screenshot (112)
Screenshot (113)
Screenshot (114)
Screenshot (115)
Screenshot (116)

### CODE REVIEW

**Review by nimrit-singh — COMMENTED** [review]
reviewed 01-10-2024

**Review by nimrit-singh — APPROVED** [review]
reviewed

File: package.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     "jsdom": "^21.1.1",
     "local-ssl-proxy": "^2.0.5",
     "pre-commit": "^1.2.2",
-    "prettier": "2.3.1",
+    "prettier": "^2.8.8",
```
Comment (vinit717) [inline package.json:32]: Why did you update the package json?

**Review by vinit717 — CHANGES_REQUESTED** [review]
Please write the test and why there is both package lock and yarn lock ?

File: package.json (hunk 1 above)
Comment (shantanu-02) [reply package.json]: Did it by mistake....resolved all the comments. Please review it

File: footerTest.js
Comment (vinit717) [inline footerTest.js:1]: Is this test related to the changes you made?

Comment (shantanu-02) [reply footerTest.js:1]: Yes sir

**Comment by vinit717** [comment]
The footer is not aligned at applications and discord group page

File: __tests__/applications/applications.test.js
Comment (vinit717) [inline __tests__/applications/applications.test.js:1]: Use data test-id instead of class/id

File: groups/style.css [hunk 2]
```diff
@@ -100,6 +100,7 @@ body {
   justify-content: center;
   align-items: center;
   flex-direction: column;
+  min-height: 98vh;
```
Comment (vinit717) [inline groups/style.css:103]: Where 98 come from is UI is working on all the different size

File: applications/style.css [hunk 3]
```diff
@@ -46,6 +46,13 @@ body {
   text-align: center;
 }
 
+.footer {
+  width: 100vw;
+  text-align: center;
+  position: fixed;
+  bottom: 5px;
```
Comment (vinit717) [inline applications/style.css:53]: Why here 5px and other place 100ox

File: footerTest.js [hunk 4]
```diff
@@ -18,6 +21,9 @@ let urls = [
   'https://dashboard.realdevsquad.com/featureflag/index.html',
   'https://dashboard.realdevsquad.com/wallet/index.html',
   'https://dashboard.realdevsquad.com/online-members/online-members.html',
+  // 'https://dashboard.realdevsquad.com/users/discord/index.html',
+  // 'https://dashboard.realdevsquad.com/groups/index.html',
+  // 'https://dashboard.realdevsquad.com/applications/index.html',
 ];
```
Comment (vinit717) [inline footerTest.js:27]: remove the comments

**Review by vinit717 — CHANGES_REQUESTED** [review]
fix the build on CI

File: __tests__/applications/applications.test.js
Comment (shantanu-02) [reply __tests__/applications/applications.test.js:1]: Have done the changes according to data-test-id

File: groups/style.css (hunk 2 above)
Comment (shantanu-02) [reply groups/style.css]: Fixed it.

File: applications/style.css (hunk 3 above)
Comment (shantanu-02) [reply applications/style.css:66]: Fixed it as well.

File: footerTest.js (hunk 4 above)
Comment (shantanu-02) [reply footerTest.js]: Done

File: __tests__/applications/applications.test.js
Comment (vinit717) [inline __tests__/applications/applications.test.js:1]: remove the comment from tests

File: applications/style.css (hunk 3 above)
Comment (vinit717) [reply applications/style.css:66]: the numbers should be multiple of 4 as per design system

**Review by vinit717 — CHANGES_REQUESTED** [review]
Can you add footer to different and import at every other place where it is used

File: __tests__/applications/applications.test.js
Comment (shantanu-02) [reply __tests__/applications/applications.test.js:1]: Removed and fixed the changes...making the reusable footer is possible but then I will have to make the changes to other previously written webpages also and will need to write tests for it entirely, which will exceed my deadline for this task, what should I do then?

File: applications/style.css (hunk 3 above)
Comment (shantanu-02) [reply applications/style.css]: Fixed it

**Comment by vinit717** [comment]
> o different and import at every other place where it is used



> Can you add the footer to a different file/folder and import it at every other place where it is used? As you are repeating same HTML several times

@shantanu-02

File: __tests__/applications/applications.test.js
Comment (shantanu-02) [reply __tests__/applications/applications.test.js:1]: I have added the reusable footer component to the code of webpages. Kindly review.

File: footer/footerComponent.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+          >
+            open sourced repo
+          </a>
+        </p>
+      </footer>
+    `;
+
+  // Insert the footer into the DOM
+  document.body.insertAdjacentHTML('beforeend', footerHTML);
+}
```
Comment (vinit717) [inline footer/footerComponent.js:23]: remove the comments

File: groups/index.html [hunk 6]
```diff
@@ -44,19 +44,13 @@
...
-          rel="noopener noreferrer"
-        >
-          open sourced repo
-        </a>
-      </p>
-    </footer>
+    <script src="../footer/footerComponent.js"></script>
+    <script>
+      document.addEventListener('DOMContentLoaded', () => {
+        const href = 'https://github.com/Real-Dev-Squad/website-dashboard';
```
Comment (vinit717) [inline groups/index.html:50]: As the href is the same at every place can you make this also into a function and put the link in constant?

File: footer/footerComponent.js (hunk 5 above)
Comment (shantanu-02) [reply footer/footerComponent.js]: Done.

File: groups/index.html (hunk 6 above)
Comment (shantanu-02) [reply groups/index.html]: done.

**Review by vinit717 — CHANGES_REQUESTED** [review]
After my last review why there too much file changed?
also update the screen shot from the pr

**Comment by shantanu-02** [comment]
> After my last review why there too much file changed? also update the screen shot from the pr

Beacuse previously there were pages which had footer...but only 3 files didn't had any...so my task ws to add fotter to those pages only...That's why asked you before making reusable component...since I made the footer component I replaced that component with previously written footers

**Comment by shantanu-02** [comment]
> After my last review why there too much file changed? also update the screen shot from the pr

Done with the new screenshots.

**Comment by vinit717** [comment]
> > After my last review why there too much file changed? also update the screen shot from the pr
> 
> Beacuse previously there were pages which had footer...but only 3 files didn't had any...so my task ws to add fotter to those pages only...That's why asked you before making reusable component...since I made the footer component I replaced that component with previously written footers

So for now update only the 3 files for rest of the file create another pr

**Comment by shantanu-02** [comment]
> > > After my last review why there too much file changed? also update the screen shot from the pr
> > 
> > 
> > Beacuse previously there were pages which had footer...but only 3 files didn't had any...so my task ws to add fotter to those pages only...That's why asked you before making reusable component...since I made the footer component I replaced that component with previously written footers
> 
> So for now update only the 3 files for rest of the file create another pr

Done with the changes please review.

File: applications/style.css [hunk 7]
```diff
@@ -59,6 +59,13 @@ body {
   text-align: center;
 }
 
+.footer {
+  width: 100vw;
+  text-align: center;
+  position: fixed;
+  bottom: 4px;
+}
+
```
Comment (vinit717) [inline applications/style.css:68]: Why you are repeating same class and property in different file

File: groups/style.css [hunk 8]
```diff
@@ -102,6 +102,11 @@ body {
   flex-direction: column;
 }
 
+.footer {
```
Comment (vinit717) [inline groups/style.css:105]: here

File: applications/style.css (hunk 7 above)
Comment (shantanu-02) [reply applications/style.css]: Fixed it

File: groups/style.css (hunk 8 above)
Comment (shantanu-02) [reply groups/style.css]: Yes, did it

File: footerTest.js [hunk 9]
```diff
@@ -1,4 +1,7 @@
 const puppeteer = require('puppeteer');
+const EventEmitter = require('events');
+
+EventEmitter.defaultMaxListeners = 15;
```
Comment (vinit717) [inline footerTest.js:4]: Why did you change this 
is this required

Comment (shantanu-02) [reply footerTest.js:4]: Yes, by default max listeners are only 10...so after adding new footers I'll have to add those as well.
Though it is not required it is related to this only

Comment (shantanu-02) [reply footerTest.js:4]: It is required for tests...It checks whether on live site footer is correctly rendered or not...But since my changes are not live yet...I will have to make another PR after this one is merged.

**Comment by vinit717** [comment]
Can you add more screenshots with different screen size also fix the failing test

File: __tests__/users/App.test.js [hunk 10]
```diff
@@ -68,6 +68,12 @@ describe('App Component', () => {
   });
 
   it('should render all sections', async () => {
+    // Wait for the sections to be rendered
```
Comment (vinit717) [inline __tests__/users/App.test.js:71]: Can you remove the comment please

File: __tests__/users/App.test.js [hunk 11]
```diff
@@ -68,6 +68,12 @@ describe('App Component', () => {
...
+    await page.waitForSelector('.tabs_section');
+    await page.waitForSelector('.users_section');
+    await page.waitForSelector('.user_card');
```
Comment (vinit717) [inline __tests__/users/App.test.js:74]: use data test id if possible

**Review by Achintya-Chatterjee — APPROVED** [review]

File: __tests__/users/App.test.js (hunk 10 above)
Comment (shantanu-02) [reply __tests__/users/App.test.js]: Done

**Review by vinit717 — APPROVED** [review]
